Can't build latest master on Visual Studio (error C2938: 'std::enable_if_t<false,int>' : Failed to specialize alias template)

### Tested versions

40cf5a0f50b4ddbd1d6990c7e7b92226665d7053

### System information

Godot v4.6.dev (207a0ac02) - Windows 10 (build 19045) - Multi-window, 2 monitors - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3080 (NVIDIA; 32.0.15.8115) - AMD Ryzen 7 3700X 8-Core Processor (16 threads) - 31.95 GiB memory

### Issue description

Trying to build the latest master branch and I get this:
```
.\core/variant/required_ptr.h(134): error C2938: 'std::enable_if_t<false,int>' : Failed to specialize alias template
.\core/variant/required_ptr.h(133): note: see reference to alias template instantiation 'std::enable_if_t<false,int>' being compiled
.\servers/physics_2d/physics_server_2d_dummy.h(106): note: see reference to class template instantiation 'RequiredResult<PhysicsDirectSpaceState2D>' being compiled
scons: *** [bin\obj\main\main.windows.editor.x86_64.obj] Error 2
scons: building terminated because of errors.
```

I've tried in Visual Studio 2022, 2019, and even installed 2026.  Tried some different msvc versions as well, but I always got that error.  Maybe something wrong on my end, but I can't figure it out.  If nothing else, this bug report might help somebody once the solution is found. 😅

### Steps to reproduce

Open the visual studio command prompt in the godot directory and type `scons p=windows`  My previous full scons command line was `scons p=windows tools=yes vsproj=yes debug_symbols=yes target=editor tracy_enable=yes -j10` but it still reproduces with the simple one.

### Minimal reproduction project (MRP)

n/a

--- a/core/variant/required_ptr.h
+++ b/core/variant/required_ptr.h
@@ -130,7 +130,7 @@ public:
 		return _value;
 	}
 
-	template <typename T_Other, std::enable_if_t<std::is_base_of_v<RefCounted, T> && std::is_base_of_v<T, T_Other>, int> = 0>
+	template <typename U = T, typename T_Other, std::enable_if_t<std::is_base_of_v<RefCounted, U> && std::is_base_of_v<U, T_Other>, int> = 0>
 	_FORCE_INLINE_ operator Ref<T_Other>() const {
 		return Ref<T_Other>(_value);
 	}

